### ISSUE TITLE
SUGGESTION: Synology support

### ISSUE THREAD

**Issue by mike-ferenduros** [beville/ComicStreamer#15]
I've had a couple of requests for this, so passing them on: ComicStreamer seems like a natural fit to run on NASes, so it'd be nice to have a pre-built package for Synology devices (or whatever other platforms make sense).

**Comment by timbueno** [comment]
I second this. Came across this program while looking to host my comic collection on my Synology.

**Comment by alibolcakan** [comment]
+1

**Comment by JSJarvis** [comment]
I agree. Having NAS distributions would be ideal. I have a Drobo and just came across the Chunky Comic Reader today that will work with ComicStreamer. I would love to have it running on my NAS so I didn't have to load comics to my iPad or iPhone to read them.

**Comment by DeusMaximus** [comment]
I did up a quick list of prerequisites for FreeBSD, which should be handy if you're running ComicStreamer from a FreeNAS jail:
https://github.com/beville/ComicStreamer/wiki/FreeBSD-requirements

**Comment by davide-romanini** [comment]
It basically works on my Zyxel NSA-325 with ffp. There are some problems though but nothing unsolvable.

**Comment by davide-romanini** [comment]
Successfully serving ~1500 on my Zyxel using my fork, it definitely works but requires some more fine tuning on NAS.

**Comment by bonebox** [comment]
The main issue I found if you're on a Synology NAS is it does not provide a default libwebp implementation.  Therefore ComicStreamer won't start since it requires PIL built with webp support.  So you have to:
1. Cross-compile libwebp from source for your specific Synology architecture (I did it on a Debian virtual machine)
2. Manually add the libwebp.so files to /lib on your NAS
3. Reinstall Pillow via pip on your NAS so that it builds with webp support.  

The other ComicStreamer dependencies can be installed directly on the NAS via pip.  

The only other issue I recall is ComicStreamer can't find the libunrar.so library, so you have to `export UNRAR_LIB_PATH=/lib/libunrar.so` before running comicstreamer.

Anyway, I have it successfully running on my DS414 using @davide-romanini's fork.  Now if it just supported a reverse-proxy path so I can access it through my SSL port...

**Comment by davide-romanini** [comment]
Thank for your report! I think we should do a better graceful degradation with somewhat optional dependencies. CS should start at least with minimal cbz/jpg support and give the user information for adding webp/unrar/whatelse. 
Another, perhaps better option, is to improve automatic build and package all dependencies even for NAS platforms. Maybe you can help providing details about your cross-compile virtual machine. Eventually I can try to add a Vagrant configuration to the project to gradually automate the process.

**Comment by bonebox** [comment]
I'd be glad to provide any details that would be helpful.  There is nice a project here https://github.com/SynoCommunity/spksrc for aiding in building native Synology packages for various architectures.  It could possibly help simplify the process, at least for Synology systems...

**Comment by rousseauxy** [comment]
@bonebox you don't happen to have it written down somewhere how to do the steps above? :)
Or has there been any progress in making this synology friendly?

**Comment by Lasborg** [comment]
I hope that there are more work being done on getting synology support for comicstreamer. Or at least that someone can make a detailed tutorial on how to install.
Right now i have installed Ubooquity, but it does not work as well as i would like.

**Comment by Finch106** [comment]
Anyway to automatically trigger a scan when a comic is placed on an NAS? Does anyone have a script that could do that?
